src/jit/lower.cpp

```diff
@@ -924,6 +924,7 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
             {
                 var_types regType          = fieldListPtr->gtGetOp1()->TypeGet();
                 argSplit->m_regType[index] = regType;
+                fieldListPtr->gtRegNum     = REG_NA;
```

Comment:
Since you are clearing the register assignment here, is it necessary to do so in `LowerArg()`?

Reply:
As you say, I'll remove clearing in `LowerArg()`. It's better to handle remain regression in other issue or pr.